File: server/src/main/java/com/koushik/movieflix/controller/CommentController.java
package com.koushik.movieflix.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.koushik.movieflix.entities.CommentEntity;
import com.koushik.movieflix.service.CommentService;

/**
 * @author Koushik
 *
 */
@RestController
public class CommentController {
	
	@Autowired
	private CommentService service;
	
	@RequestMapping(method=RequestMethod.GET,value="/comments")
	public List<CommentEntity> showAll(){
		return service.showAll();
	}

	@RequestMapping(method=RequestMethod.GET,value="/comments/title/{id}")
	public List<CommentEntity> showCommentByMovie(@PathVariable("id") String mid){
		return service.showCommentsByMovie(mid);
	}
	
	@RequestMapping(method=RequestMethod.POST,value="comments/user/{uid}/title/{mid}")
	public CommentEntity createComment(@PathVariable("uid") String userId,@PathVariable("mid") String movieId,@RequestBody CommentEntity entity){
		return service.createComment(userId, movieId, entity); 
	}

}

File: client/src/app/movies/comment.controller.js
/**
 * Created by Koushik on 12/28/2016.
 */

(function () {
    angular.module('movieflix')
        .controller('CommentController',CommentController);
    CommentController.$inject = ['CommentService','$cookies'];
    function CommentController(CommentService,$cookies){

        var cctl = this;
        this.commentMovie = function (movieid,comment) {
            console.log(movieid,comment);
            var userid = $cookies.get('userId');
            CommentService.comment(userid,movieid,comment)
                .then(function (resp) {
                    cctl.comments = resp;
                },function (err) {
                    console.log(err);
                })

        }

    }
})();